backend/src/services/model.ts

```diff
@@ -262,19 +275,33 @@ export async function updateModelCard(
   return revision
 }
 
-export type UpdateModelParams = Pick<
-  ModelInterface,
-  'name' | 'description' | 'visibility' | 'collaborators' | 'settings'
->
-export async function updateModel(user: UserInterface, modelId: string, diff: Partial<UpdateModelParams>) {
+export type UpdateModelParams = Pick<ModelInterface, 'name' | 'description' | 'visibility' | 'collaborators'>
+export async function updateModel(
+  user: UserInterface,
+  modelId: string,
+  modelDiff: Partial<UpdateModelParams>,
```

Comment:
Why not include settings in the model?  What makes settings special?

Reply:
Ah.  Presumably because we only use `Object.assign()` below, which isn't recursive.  I would prefer a recursive merging here, as opposed to treating 'settings' as a special case.  I think something like:

```
.mergeWith({}, model, modelDiff, (a, b) => 
  _.isArray(b) ? b : undefined
);
```

Might work.